protocols/mempool/v0/mempool-v0.md

```diff
@@ -0,0 +1,169 @@
+   of entries. [[`rpc.core.mempool.unconfirmed_txs`][unconfirmed_txs]]
+
+### Update transactions in the mempool
+
+6. BlockExecutor updates the mempool after committing a block.
+    [[`state.BlockExecutor.Commit`][Commit]]
+
+    6a. Flush to ensure all async requests have completed in the ABCI app before
+        commit (see the [note below](#flush)). [[`mempool.CListMempool.FlushAppConn`][FlushAppConn]]
+
+    6b. Call mempool's `Update` for all transactions in the block. 
+
+7. Update the mempool. [[`mempool.CListMempool.Update`][Update]]
+    <!-- [spec:Consensus_Update] -->
+
+    7a. If a transaction is valid, put it in the cache (in case it was missing);
+    otherwise, remove it from the cache, to give it a chance to be resubmitted
+    later.
+
+    7b. Remove from the mempool each transaction in the block.
+
+    7c. If there are still transactions in the mempool and `config.Recheck` is
+    true, then send one `CheckTx` ABCI request to the app for every transaction
+    in the block. [[`mempool.CListMempool.recheckTxs`][recheckTxs]]
+    
+    7d. If there are still transactions in the mempool and `config.Recheck` is
+    false, then notify Consensus that there are transactions available.
+        [[`mempool.CListMempool.notifyTxsAvailable`][notifyTxsAvailable]]
+
+8. On `CheckTx` response, if the transaction is invalid.
+    [[`mempool.CListMempool.resCbRecheck`][resCbRecheck]]
+    <!-- [spec:ReceiveRecheckTxResponse] -->
+    
+    8a. Remove it from the mempool.
+    
+    8b. Remove it from the cache.
+
+    8c. If there are still transactions in the mempool, notify Consensus.
+        [[`mempool.CListMempool.notifyTxsAvailable`][notifyTxsAvailable]]
+
+### Propagate validated transactions to peers
+
+9. Propagate each transaction in the mempool to all peers.
+    [[`mempool.Reactor.broadcastTxRoutine`][broadcastTxRoutine]]
+    <!-- [spec:P2P_SendTx] -->
+
+    9a. Loop through the mempool.
+        [[`mempool.CListMempool.TxsFront/TxsWaitChan`][txs-loop]]
+
+    9b. Make sure the peer is up to date; then get its height.
+        [[`peer.Get(types.PeerStateKey).(PeerState)`][Peer.Get]]
+      - If we suspect that the peer is lagging behind, wait some time before
+        checking again if the peer has caught up. See
+        [RFC-103](https://github.com/cometbft/cometbft/blob/main/docs/rfc/rfc-103-incoming-txs-when-catching-up.md)
+
+    9c. Send all transactions in the mempool to the peer, one by one.
+
+## Links to code
+### RPC
+- [broadcast_tx_*]
+- [unconfirmed_txs]
+
+### Mempool reactor
+- [Receive]
+- [broadcastTxRoutine]
+
+### CListMempool
+- [CheckTx]
+- [Update]
+- [TxsFront/TxsWaitChan][txs-loop]
+- [FlushAppConn]
+- [resCbFirstTime]
+- [resCbRecheck]
+- [recheckTxs]
+- [notifyTxsAvailable]
+
+### Consensus
+- [CreateProposalBlock]
+- [Commit]
+- [PeerState]
+
+### Peer
+
+- [Peer.Get]
+
+[broadcast_tx_*]: https://github.com/cometbft/cometbft/blob/v0.38.x/rpc/core/mempool.go#L22-L144
```

Comment:
I'd recommend linking to files at a specific commit instead of using the branch reference, otherwise the links may go out of sync.

Reply:
Thanks, I updated the links. At the time of writing there was no v0.38 release.

Reply:
Great, a release tag also works!